### DISCUSSION TITLE
[Greenfield API] Include amount and type of crypto received in GET invoice(s)

### DISCUSSION THREAD

**Discussion by Atitlanio** [btcpayserver/btcpayserver#4656] [Ideas / Feature Requests]
When using the API to retrieve invoices, it would be nice to have more information, particularly **_the amount and type of crypto received_**. At the moment it only provides the amount of fiat received, whereas exporting the invoices into csv/JSON provides much more detailed information.

### Comparison
**1) Greenfield response:**
```
{
    "metadata": {},
"checkout": {},
"receipt": {},
"id": "string",
"storeId": "string",
"amount": "5.00",
"currency": "USD",
"type": "Standard",
"checkoutLink": "string",
"createdTime": 1592312018,
"expirationTime": 1592312018,
"monitoringExpiration": 1592312018,
"status": "Expired",
"additionalStatus": "Invalid",
"availableStatusesForManualMarking": 

    [],
    "archived": true

}
```
[Greenfield Ref](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoices)

**2) Exporting csv/JSON via the web:**
```
  {
    "ReceivedDate": "2023-01-16T00:14:10Z",
    "StoreId": "redacted",
    "OrderId": "",
    "InvoiceId": "redacted",
    "InvoiceCreatedDate": "2023-01-14T00:12:33Z",
    "InvoiceExpirationDate": "2023-01-14T00:17:33Z",
    "InvoiceMonitoringDate": "2023-01-16T00:17:33Z",
    "PaymentId": "redacted",
    "Destination": "redacted",
    "PaymentType": "Off-Chain",
    "CryptoCode": "BTC",
    "Paid": "0.00009418",
    "NetworkFee": "0.0",
    "ConversionRate": 191137.15,
    "PaidCurrency": "18.001",
    "InvoiceCurrency": "GTQ",
    "InvoiceDue": -0.001,
    "InvoicePrice": 18.0,
    "InvoiceItemCode": null,
    "InvoiceItemDesc": null,
    "InvoiceFullStatus": "Settled",
    "InvoiceStatus": "complete",
    "InvoiceExceptionStatus": "",
    "BuyerEmail": null,
    "Accounted": true
  },
```

**Comment by bolatovumar** [comment]
Looks like what we are missing primarily is the data around invoice payments. When exporting CSV or JSON it's all there but Greenfield API doesn't provide any of that data on the invoice.

I can look into adding this data to Greenfield API endpoint.

**Reply by bolatovumar** [reply]
PR started: https://github.com/btcpayserver/btcpayserver/pull/4681

**Reply by bolatovumar** [reply]
Hi @Atitlanio, I closed my PR because I believe you can use the https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoicePaymentMethods endpoint to get the data you need. Let me know if there is something missing that you would like to see.

**Comment by Atitlanio** [comment]
Hmm, so the application that I am using it is for a restaurant environment. The customers selected items off of a front-end menu. When they go to checkout, they can select to add a tip for the employees. They hit "Pay" which sends a [POST request](https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_CreateInvoice) for an invoice with the payload amount which essentially boils down to :

```
{
    ... ,
    amount: if ( tipAmount ) { cartTotal + tipAmount } else { cartTotal } ,
    ... ,
}
```

So in order to separate the tips from the total revenue, I am running a script that goes through all of the invoices and tallies tips versus the invoice "subtotal" which would be without the tip.

So the relevant issue then is that I'm calculating tips and revenue in time-dependent fiat price values and don't reflect their value today as a function of the exchange rate for BTC.

To me, it makes more sense that the invoice data would prioritize showing the amount of the currency that it received, which in this case would be BTC! That way its value is always what it is, the amount of BTC. If it's shown in the Fiat value that it was converted to, then the value doesn't really mean much unless you have the rate at the time.

I can certainly get it from the endpoint you referenced, but as I loop through the entirety of invoices, it would be burdensome to do an API request for each one of them to obtain the BTC value. 

edit: meant to say POST request to create an invoice!
